portality/models/account.py

```diff
@@ -92,7 +94,11 @@ def set_password(self, password):
         self.data['password'] = generate_password_hash(password)
 
     def check_password(self, password):
-        return check_password_hash(self.data['password'], password)
+        try:
+            return check_password_hash(self.data['password'], password)
+        except:
```

Comment:
this is an undesirable bare exception (it should be `KeyError`), and I think a 401 or 403 might be more appropriate in this case.

Reply:
True. Good lesson. I haven't thought about it.